### PR TITLE
Implement evidence submission process, #4195

### PR DESCRIPTION
**Pull request by hyifeng**

### CODE REVIEW

File: packages/next-common/components/fellowship/core/members/actions/more/submitEvidenceItem/popup.jsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    }
+
+    const hexDigest = "0x" + Buffer.from(digest).toString("hex");
+    return api.tx.fellowshipCore?.submitEvidence(wish, hexDigest);
+  }, [api, address, upload, evidence, wish, dispatch]);
+
+  const fnWaitSync = useWaitSyncBlock("Evidence submitted", () => {
+    dispatch(incFellowshipCoreMembersTrigger());
+  });
+  const onSubmit = (_, blockHash) => blockHash && fnWaitSync(blockHash);
```
Comment (wliyongfeng) [inline packages/next-common/components/fellowship/core/members/actions/more/submitEvidenceItem/popup.jsx:85]: Why do we need update core members after evidence submission?

File: packages/next-common/components/fellowship/core/members/actions/more/submitEvidenceItem/popup.jsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        />
+        <Editor
+          value={evidence}
+          onChange={setEvidence}
+          contentType={"markdown"}
+          setContentType={() => {}}
+          loadSuggestions={() => []}
+          minHeight={100}
+          previewerPlugins={[]}
+          setQuillRef={() => {}}
```
Comment (wliyongfeng) [inline packages/next-common/components/fellowship/core/members/actions/more/submitEvidenceItem/popup.jsx:129]: Can we ignore attributes with empty functions? Are there default values in `Editor` component?

File: packages/next-common/components/fellowship/core/members/actions/more/submitEvidenceItem/popup.jsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        />
+      </div>
+
+      <TxSubmissionButton
+        title="Save & Publish"
+        loadingText={uploading ? "Saving..." : "Publishing..."}
```
Comment (wliyongfeng) [inline packages/next-common/components/fellowship/core/members/actions/more/submitEvidenceItem/popup.jsx:135]: 1. Change "Saving" to "Uploading to IPFS".
2. Change "Publishing" to "Submitting".

File: packages/next-common/components/fellowship/core/members/evidence.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import { useEffect, useState } from "react";
+import { useContextApi } from "next-common/context/api";
+import { textEllipsis } from "next-common/utils";
+import getIpfsLink from "next-common/utils/env/ipfsEndpoint";
+import { create as createDigest } from "multiformats/hashes/digest";
+import { CID } from "multiformats";
+import { hexToU8a } from "@polkadot/util";
+import Tooltip from "next-common/components/tooltip";
+
+function useFellowshipCoreMemberEvidence(address) {
```
Comment (wliyongfeng) [inline packages/next-common/components/fellowship/core/members/evidence.js:13]: How about `useSubFellowshipCoreMemberEvidence` considering we are subscribing a storage?

File: packages/next-common/components/fellowship/core/members/evidence.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        setLoading(false);
+      });
+  }, [api, address]);
+
+  return {
+    wish,
+    evidence,
+    loading,
+  };
+}
```
Comment (wliyongfeng) [inline packages/next-common/components/fellowship/core/members/evidence.js:49]: Please put this hook in a separated file.

File: packages/next-common/components/fellowship/core/members/evidence.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+          rel="noreferrer"
+          href={getIpfsLink(cid)}
+        >
+          {textEllipsis(cid, 4, 4)}
+        </a>
+      </div>
+    );
+  } else {
+    content = <span className="text-textTertiary">-</span>;
+  }
```
Comment (wliyongfeng) [inline packages/next-common/components/fellowship/core/members/evidence.js:79]: Write `} else if (!evidence) {` first, so we can leverage this short logic.

File: packages/next-common/lib/button/index.jsx [hunk 7]
```diff
@@ -52,7 +53,16 @@ export default function _Button(props) {
...
+        loadingText ? (
+          <>
+            <span className="inline-flex mr-1">
+              <SystemLoading className={cn(smallSize && "w-4 h-4")} />
+            </span>
+            {loadingText}
+          </>
+        ) : (
+          <SystemLoading className={cn(smallSize && "w-4 h-4")} />
+        )
```
Comment (wliyongfeng) [inline packages/next-common/lib/button/index.jsx:65]: Please confirm this modification with @2nthony . I'm not sure whether this change should be implemented in `button` component. I think we can implement another alternative button on higher business components.

File: packages/next-common/components/fellowship/core/members/actions/more/submitEvidenceItem/popup.jsx [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+    let digest;
+    try {
+      digest = CID.parse(cid).toV0().multihash.digest;
+    } catch (e) {
+      dispatch(newErrorToast("Failed to parse CID digest"));
+      return;
+    }
+
+    const hexDigest = "0x" + Buffer.from(digest).toString("hex");
```
Comment (wliyongfeng) [inline packages/next-common/components/fellowship/core/members/actions/more/submitEvidenceItem/popup.jsx:78]: Does this code work? `Buffer` is only available in node.js. What's the type of `digest`? Did you try passing `digest` directly to `submitEvidence`?

Comment (hyifeng) [reply packages/next-common/components/fellowship/core/members/actions/more/submitEvidenceItem/popup.jsx:78]: Yes it works, and there is an error when try to pass `digest` directly to `submitEvidence`,

File: packages/next-common/components/fellowship/core/members/actions/more/submitEvidenceItem/popup.jsx (hunk 1 above)
Comment (hyifeng) [reply packages/next-common/components/fellowship/core/members/actions/more/submitEvidenceItem/popup.jsx]: refactor to use subscribe data instead

**Pull request merged**